### PR TITLE
Fix: env0_template opentofu_version pattern match failure for RESOLVE…

### LINKED ISSUE #814: env0_template opentofu_version pattern match failure for RESOLVE_FROM_CODE
**Describe the bug**
When setting the opentofu version to resolve from code in the ui and trying to update the env0_template resource config to match remote config, it fails when the opentofu_version attribute is set to "RESOLVE_FROM_CODE" to match remote config.

**To Reproduce**
Steps to reproduce the behavior.

1- update the env0 template to use opentofu and resolve from code for version
2 - run plan which shows the drift
3- update config to match remote
4- pattern match failure for "RESOLVE_FROM_CODE"

```
resource "env0_template" "this" {
    id                    = "d0f1ac44-33fc-4555-9999-00000000"
    name                  = "terraform-workspace"
    opentofu_version      = "RESOLVE_FROM_CODE" -> null
    type                  = "opentofu" -> "terraform"
    # (13 unchanged attributes hidden)
}
```

```
│ Error: must match pattern ^(?:[0-9]\.[0-9]{1,2}\.[0-9]{1,2})|1\.6\.0-alpha$
│
│   with module.env0_template_project_environment.env0_template.this,
│   on modules/env0_template_project_environment/main.tf line 18, in resource "env0_template" "this":
│   18:   opentofu_version = "RESOLVE_FROM_CODE"
```

**Expected behavior**
"RESOLVE_FROM_CODE" should be a valid value for the opentofu_version attribute

**Provider Version**
1.17.1

### PR DESCRIPTION
**Pull request by TomerHeber**
…_FROM_CODE

### Issue & Steps to Reproduce / Feature Request
fixes #814 

### Solution

1. Fixed the regex.
2. Updated documentation.
3. Updated acceptance tests.
4. Updated integration tests.

### CODE REVIEW

File: env0/resource_template.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func getTemplateSchema(prefix string) map[string]*schema.Schema {
 		},
 		"opentofu_version": {
 			Type:             schema.TypeString,
-			Description:      "the Opentofu version to use (example: 0.36.5)",
+			Description:      "the Opentofu version to use (example: 1.6.2). Setting to 'RESOLVE_FROM_CODE' extracts the version from the terraform code during runtime. Setting to `latest`, the version used will be the most recent one available for Opentofu.",
```
Comment (chpl) [inline env0/resource_template.go:178]:
```suggestion
			Description:      "the Opentofu version to use (example: 1.6.2). Setting to 'RESOLVE_FROM_CODE' extracts the version from the Opentofu code during runtime. Setting to `latest`, the version used will be the most recent one available for Opentofu.",
```

File: env0/validators.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func NewGreaterThanValidator(greaterThan int) schema.SchemaValidateDiagFunc {
 }
 
 func NewOpenTofuVersionValidator() schema.SchemaValidateDiagFunc {
-	return NewRegexValidator(`^(?:[0-9]\.[0-9]{1,2}\.[0-9]{1,2})|1\.6\.0-alpha$`)
+	return NewRegexValidator(`^(?:[0-9]\.[0-9]{1,2}\.[0-9]{1,2})|RESOLVE_FROM_CODE|latest$`)
```
Comment (chpl) [inline env0/validators.go:119]: should still support alpha and beta versions

Comment (TomerHeber) [reply env0/validators.go:119]: it's supported. The regex does not end with a $. Therefore, these are matching.

Comment (chpl) [reply env0/validators.go:119]: then you should drop the `$` in the `latest$`

Comment (chpl) [reply env0/validators.go:119]: or maybe `(?:^[0-9]\.[0-9]{1,2}\.[0-9]{1,2}(-.+)?$)|^RESOLVE_FROM_CODE$|^latest$`

Comment (chpl) [reply env0/validators.go:119]: @TomerHeber ^

Comment (TomerHeber) [reply env0/validators.go:119]: `(?:^[0-9]\.[0-9]{1,2}\.[0-9]{1,2}(?:-.+)?$)|^RESOLVE_FROM_CODE$|^latest$`

Comment (TomerHeber) [reply env0/validators.go]: updated.

**Review by chpl — APPROVED** [review]

**Pull request merged**